File: ArraysAndHashing/TopKFrequent.py
class Solution(object):
    def topKFrequent(self, nums, k):
        m = {}
        for n in nums:
            m[n] = m.get(n,0) + 1
        while len(m) > k:
            m.pop(min(m,key=m.get))
        return m.keys()
    
# Time complexity: O(n)
# Space complexity: O(n)

File: ArraysAndHashing/TopKFrequent.cpp
#include <vector>
#include <map>
#include <queue>
#include <iostream>
    
using namespace std;

class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        map<int,int> m;
        for(int it = 0 ; it < nums.size() ; it++)
            m[nums[it]]++;
        int max = 0;
        int num = 0;
        vector<int> result;
        while(k > 0)
        {
            for(auto it = m.begin(); it != m.end(); it++)
            {
                if(it->second > max)
                {
                    max = it->second;
                    num = it->first;
                }
            }
            max = 0;
            result.push_back(num);
            m.erase(num);
            k--;
        }
        return result;
    }
};

//this solution has a time complexity of O(n) and space complexity of O(n) using map